src/app/drive/components/FileViewer/FileViewer.tsx

```diff
@@ -39,7 +39,10 @@ interface FileViewerProps {
   totalFolderIndex?: number;
   fileIndex?: number;
   dropdownItems?: TopBarActionsMenu;
-  renameItemFromKeyboard?: () => void;
+  keyboardShortcuts?: {
+    renameItemFromKeyboard: ((item) => void) | undefined;
```

Comment:
rather than defining each handler like this, why dont you create something like a `KeyboardShortcut` object that looks something like this:

```
type KeyboardShortcut {
    key: string;
    startListening: () => void;
    stopListening: () => void;
}
```

You can then create a class that encapsulates all that logic, and control how it behaves from each component, in terms of props, you just pass an array of `KeyboardShortcut`